src/deepsparse/pipeline.py

```diff
@@ -328,6 +331,9 @@ def run(
                 inference_state = graph.inf
                 next_step = graph.step
 
+        timer = inference_state.get_state("timer")
```

Comment:
let's add a timer property to inference state as discussed to handle the optional timer